File: LB-Assignments/Assignment_36/program3.cpp
/*
Accept number from user and toggle 7th bit of number
*/
#include<iostream>
using namespace std;

typedef unsigned int UINT;

UINT ToggleBit(UINT iNo)
{
    UINT iMask = 0x40;
    UINT iAns = 0;

    iAns = iNo ^ iMask;

    return iAns;
}

int main()
{
    UINT iValue = 0;
    UINT iRet = 0;

    cout<<"ENter number : \n";
    cin>>iValue;

    iRet = ToggleBit(iValue);

    cout<<"Modified number is : "<<iRet<<"\n";
    
    return 0;
}

File: LB-Assignments/Assignment_37/program3.java
/*
Accept number from user and position 
and on that bit and return that number 
*/
import java.util.*;

class Bitwise
{
    public int iNo, iPos;

    public Bitwise(int a, int b)
    {
        this.iNo = a;
        this.iPos = b;
    }

    public int OnBit()
    {
        int iMask = 0x1, iAns = 0;

        iMask = iMask << (iPos -1);

        iAns = iNo | iMask;

        return iAns;
    }
}
class program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iLocation = 0, iRet = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter Location : ");
        iLocation = sobj.nextInt();

        Bitwise bobj = new Bitwise(iValue, iLocation);

        iRet = bobj.OnBit();

        System.out.println("Modified number is : "+iRet);

    }
}